File: app/src/main/java/com/github/teranes10/androidutils/helpers/ui/Loading.java
package com.github.teranes10.androidutils.helpers.ui;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.widget.TextView;

import com.github.teranes10.androidutils.R;

public class Loading {
    private static Dialog dialog;

    private static void init(Context context) {
        dialog = new Dialog(context, R.style.FullScreenAlert);
        dialog.setContentView(R.layout.custom_loading);
        dialog.setCancelable(false);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
    }

    public static void start(Context context) {
        if (dialog == null) {
            init(context);
        }

        TextView textView = dialog.findViewById(R.id.loading_text);
        textView.setText("");
        dialog.show();
    }

    public static void start(Activity context, String message) {
        if(context.hasWindowFocus()){
            return;
        }
        if(context.isFinishing()){
            return;
        }

        if (dialog == null) {
            init(context);
        }


    }

    public static void stop() {
        if (dialog != null) {
            dialog.dismiss();
        }
    }
}

File: app/src/main/java/com/github/teranes10/androidutils/ui/Loading.kt
package com.github.teranes10.androidutils.ui

import android.app.Activity
import android.app.Dialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.View
import android.widget.LinearLayout.LayoutParams
import android.widget.TextView
import com.github.teranes10.androidutils.R

class Loading(private val context: Activity, private val fullScreen: Boolean = false) {

    private val dialog: Dialog = Dialog(context, if (fullScreen) R.style.FullScreenAlert else 0).apply {
        setContentView(R.layout.custom_loading)
        setCancelable(false)
        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
    }

    private val textView: TextView = dialog.findViewById(R.id.loading_text)

    fun start(message: String? = "") {
        if (context.isFinishing || context.isDestroyed || dialog.isShowing) {
            return
        }

        if (!message.isNullOrBlank()) {
            textView.text = message
            textView.visibility = View.VISIBLE

            val screenWidth = context.resources.configuration.smallestScreenWidthDp
            textView.textSize = when {
                screenWidth > 720 -> 16.0f
                screenWidth > 600 -> 14.0f
                else -> 12.0f
            }
        } else {
            textView.visibility = View.GONE
        }

        dialog.show()
    }

    fun stop() {
        if (dialog.isShowing) {
            dialog.dismiss()
        }
    }
}

